### PR TITLE
Issue #3013653 by ribel: Add content tagging filter to the all search page

### PR DESCRIPTION
**Pull request by ribel**
## Problem
As a user, I want to be able to see everything that relates to a certain tag.

## Solution
- Add the content tagging filters to All search
- Make sure it applies to content and groups

## Issue tracker
- https://www.drupal.org/node/3013653
- https://jira.goalgorilla.com/browse/HGT-23

## HTT
- [ ] Check out the code changes
- [ ] Enable content tagging and create a hierarchy in the vocabulary
- [ ] Tag some content and groups
- [ ] Go to the Search All page and notice you can also filter on tags (and on entity types)

## BONUS
Since I needed to add the tagging field to the filter bock on the search all page, I first needed to create a filter block. Therefore I added an entity type filter on the all search page.

## Release notes
<describe the release notes>

### CODE REVIEW

File: modules/social_features/social_tagging/social_tagging.install [hunk 1]
```diff
@@ -59,6 +67,21 @@ function social_tagging_update_8002() {
   }
 }
 
+/**
+ * Toggle all index.
+ */
+function social_tagging_update_8003() {
```
[the remark this answers is not in the document]
Comment (ribel) [reply modules/social_features/social_tagging/social_tagging.install]: @bramtenhove I have used the same approach as in PR #1094. Should I change it?

**Comment by jaapjan** [comment]
FYI there are some search tests failing on PAC

**Comment by jaapjan** [comment]
As discussed in sprint review the entity type filter on top might cause confusion for users and extra unneeded complexity. I think we need to explore different solutions for that.

By the way, not sure why you created the entity type filter, but there is also a profile tag functionality. So in theory when all these entities can have tags we can use the same (custom) filter or no?

**Comment by ronaldtebrake** [comment]
Thanks for the info. Completely agree, updated the milestone to 4.1.

**Comment by jaapjan** [comment]
when do you plan to finish this PR?

**Comment by ribel** [comment]
@jaapjan Work on it is actually already finished. This feature is also already used by HGT.
There were some tests failing, but they look like false positive, so I have updated the branch.

Regarding your question above - it was more technical decision discussed with @jochemvn and @bramtenhove during refinement. 
The same as for search content and search groups in PR #1094 - we need to have some exposed filter to have the ability to alter it and add tag filters. Since we didn't have any filter on search all page yet, we agreed that it will not harm to have filter by entity type here.

**Comment by bramtenhove** [comment]
@ribel,
cc @jochemvn, @jaapjan,

> As discussed in sprint review the entity type filter on top might cause confusion for users and extra unneeded complexity. I think we need to explore different solutions for that.
> 
> By the way, not sure why you created the entity type filter, but there is also a profile tag functionality. So in theory when all these entities can have tags we can use the same (custom) filter or no?

I think this still requires a solution. I recall more people during the review were not convinced this was a good approach.
Could this be something you can work on and discuss within the team in the 8 hours allocated for Distro work this sprint?

**Comment by jaapjan** [comment]
@ribel did you see the comment from Bram?

**Comment by ribel** [comment]
@jaapjan @bramtenhove 
We were not able to discuss it with a team yet, due to the high priority of other ongoing tasks, and also current absence half of a team.

But I still think that this will be more complex to have tagging filter there without any other exposed filter due to the architecture of social tagging module, and fact that we split one taxonomy in several filters. That's why we put entity type filter there. 

Regarding MOSS tests I see that ECI Demo and Foraus pass well, and tests for example on CBI and Philanthropy are not related to search functionality and looks like false positive.

**Comment by ribel** [comment]
@jaapjan @bramtenhove 
If you really think we don't need Entity type filter there I can do workaround and add filter there, but later hide it with code:

```
    if ($form['#id'] == 'views-exposed-form-search-all-page') {
      if (isset($form['type'])) {
        $form['type']['#access'] = FALSE;
      }
    }
```

Let me know what approach you think is better.

**Comment by bramtenhove** [comment]
@ribel, @jaapjan,
After thinking some more about this I suggest we maybe should not add the content tag filter to the distro right now. The value is not very high and adding the entity type filter reduces the value of the search tabs.

The work around you propose is adding some dead code which we'll probably never take out anymore. I think that's not a good idea.

If we do not add it to the distro I suspect we'll have to do something for the project where this is already being used as a patch. Let me know what you think and I'll create an internal ticket to follow up on this.

**Comment by jaapjan** [comment]
I'm ok with that, if @ribel agrees (closing PR for now then :-) )

**Comment by ribel** [comment]
@bramtenhove  @ribel 
If you think we don't need this in a distro - I am fine with that. But maybe we will need to raise this question if we add content tagging for profiles.
Currently, it was used only on HGT and GPEDC projects.
I will replace it with project specific code for GPEDC, and maybe you need to do the same for HGT, @bramtenhove.

**Pull request closed without merging**